oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/util/IndexDefinitionBuilder.java

```diff
@@ -567,21 +576,71 @@ private static Tree getOrCreateChild(Tree tree, String name){
     }
 
     static class SelectiveEqualsDiff extends EqualsDiff {
+
+        List<String> ignorablePropertiesList = ImmutableList.of(FulltextIndexConstants.PROP_NAME,FulltextIndexConstants.PROP_WEIGHT,FIELD_BOOST,
+                FulltextIndexConstants.PROP_SECURE_FACETS,FulltextIndexConstants.PROP_STATISTICAL_FACET_SAMPLE_SIZE,FulltextIndexConstants.PROP_FACETS_TOP_CHILDREN,
+                IndexConstants.QUERY_PATHS,FulltextIndexConstants.BLOB_SIZE,FulltextIndexConstants.COST_PER_ENTRY,FulltextIndexConstants.COST_PER_EXECUTION);
+
+
         public static boolean equals(NodeState before, NodeState after) {
             return before.exists() == after.exists()
                     && after.compareAgainstBaseState(before, new SelectiveEqualsDiff());
         }
 
+
         @Override
         public boolean propertyChanged(PropertyState before, PropertyState after) {
             if (IndexConstants.ASYNC_PROPERTY_NAME.equals(before.getName())){
                 Set<String> asyncBefore = getAsyncValuesWithoutNRT(before);
                 Set<String> asyncAfter = getAsyncValuesWithoutNRT(after);
                 return asyncBefore.equals(asyncAfter);
+            } else if (ignorablePropertiesList.contains(before.getName())) {
+                return true;
             }
             return false;
         }
 
+        @Override
+        public boolean propertyAdded(PropertyState after) {
+            if(ignorablePropertiesList.contains(after.getName())) {
+                return true;
+            }
+            return super.propertyAdded(after);
+        }
+
+        @Override
+        public boolean propertyDeleted(PropertyState before) {
+            if(ignorablePropertiesList.contains(before.getName())) {
+                return true;
+            }
+            return super.propertyDeleted(before);
+        }
+
+        @Override
+        public boolean childNodeAdded(String name, NodeState after) {
+            if(name.equals(FulltextIndexConstants.PROP_FACETS)) {
```

Comment:
While I don't have a good idea atm, but I don't quite like the assumption that any change under `facets` nodes only requires a refresh. Yes, it's true for currently supported properties but in the future if we add something else here that requires reindexing then it might slip away and won't be caught. I'd probably be happier if we make sure that adding `facets/foo` reindexes while adding `facets/topChildren` only refreshes.

Reply:
@catholicon  - makes sense . I have made the changes . Basically removed override for childNodeChanged and added handling in childNodeAdded and childNodeDeleted to set reindex to true in case there are any new properties apart from the current three . In case the new properties are also such that they wouldn't req re indexing - then of course we will have to make changes accordingly .

Reply:
Thanks for the change. Just a minor bit, maybe we could move the life of properties under facets at top along with other properties (or, I'd do that myself)

Reply:
@catholicon  - done . Moved the declaration on top . I also did another change - I had added [name] property for index node as ignorable but I realized this will conflict with changes in name property on property def node (which will req reindex if changed) - so I have removed the handling for name property.